after/syntax/gitlab.vim

```diff
@@ -2,32 +2,26 @@ if filereadable($VIMRUNTIME . '/syntax/markdown.vim')
   source $VIMRUNTIME/syntax/markdown.vim
 endif
 
-syntax match Date "\v\d+\s+\w+\s+ago"
-highlight link Date GitlabDate
-
-execute 'syntax match Unresolved /\s' . g:gitlab_discussion_tree_unresolved . '\s\?/'
-highlight link Unresolved GitlabUnresolved
-
-execute 'syntax match Unlinked /\s' . g:gitlab_discussion_tree_unlinked . '\s\?/'
-highlight link Unlinked GitlabUnlinked
-
-execute 'syntax match Resolved /\s' . g:gitlab_discussion_tree_resolved . '\s\?/'
-highlight link Resolved GitlabResolved
-
-execute 'syntax match GitlabDiscussionOpen /^\s*' . g:gitlab_discussion_tree_expander_open . '/'
-highlight link GitlabDiscussionOpen GitlabExpander
-
-execute 'syntax match GitlabDiscussionClosed /^\s*' . g:gitlab_discussion_tree_expander_closed . '/'
-highlight link GitlabDiscussionClosed GitlabExpander
-
-execute 'syntax match Draft /' . g:gitlab_discussion_tree_draft . '/'
-highlight link Draft GitlabDraft
-
-execute 'syntax match Username "@[a-zA-Z0-9.]\+"'
-highlight link Username GitlabUsername
-
-execute 'syntax match Mention "\%(' . g:gitlab_discussion_tree_expander_open . '\|'
-  \ . g:gitlab_discussion_tree_expander_closed . '\)\@<!@[a-zA-Z0-9.]*"'
-highlight link Mention GitlabMention
+let expanders = '^\s*\%(' . g:gitlab_discussion_tree_expander_open . '\|' . g:gitlab_discussion_tree_expander_closed . '\)'
+let username = '@[a-zA-Z0-9.]\+'
+
+" Covers times like '14 days ago', 'just now', as well as 'October  3, 2024'
+let time_ago = '\d\+ \w\+ ago'
+let formatted_date = '\w\+ \{1,2}\d\{1,2}, \d\{4}'
+let date = '\%(' . time_ago . '\|' . formatted_date . '\|just now\)'
+
+let published = date . ' \%(' . g:gitlab_discussion_tree_resolved . '\|' . g:gitlab_discussion_tree_unresolved . '\|' . g:gitlab_discussion_tree_unlinked . '\)\?'
+let state = ' \%(' . published . '\|' . g:gitlab_discussion_tree_draft . '\)'
+
+execute 'syntax match GitlabNoteHeader "' . expanders . username . state . '" contains=GitlabDate,GitlabUnresolved,GitlabUnlinked,GitlabResolved,GitlabExpander,GitlabDraft,GitlabUsername'
+
+execute 'syntax match GitlabDate "' . date . '" contained'
+execute 'syntax match GitlabUnresolved "' . g:gitlab_discussion_tree_unresolved . '" contained'
+execute 'syntax match GitlabUnlinked "' . g:gitlab_discussion_tree_unlinked . '" contained'
```

Comment:
The unlinked highlights are not working on this branch in the discussion tree

Reply:
In what sense does it not work? When I run neovim with the following config it works just fine, i.e., the `:Inspect` command prints `GitlabNoteHeader` and `GitlabUnlinked` :
```lua
-- This is a minimal init.lua that is used for debugging
local root = vim.fn.stdpath("run") .. "/nvim/mininit"
vim.fn.mkdir(root, "p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  {
    "jakubbortlik/gitlab.nvim",
    branch = "fix-syntax-in-discussion-tree",
    dependencies = {
      "MunifTanjim/nui.nvim",
      "nvim-lua/plenary.nvim",
      "sindrets/diffview.nvim",
      "stevearc/dressing.nvim",   -- Recommended but not required. Better UI for pickers.
      "nvim-tree/nvim-web-devicons" -- Recommended but not required. Icons in discussion tree.
    },
    enabled = true,
    build = function() require("gitlab.server").build(true) end, -- Builds the Go binary
    config = true,
  },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

Highlighting does not work when I change the symbol for unlinked comments like this:
```lua
opts = {discussion_tree = {unlinked = "x"}},
```
But this is not a problem of this branch but of the way how the symbols are hardcoded in `lua/gitlab/colors.lua` lines 9-12. I'll push a fix for this in a moment.

If the problem still persists, please let me know what it looks like how I can reproduce it.